lib/node_modules/@stdlib/plot/unicode/static_plots/bar/main.js

```diff
@@ -0,0 +1,466 @@
+		elm = new Array( 3 );
+		elm[ 1 ] = 0;
+		elm[ 0 ] = xpos;
+		elm[ 2 ] = i.toString();
+		y.push( elm );
+		i += step;
+		i = round( i * 10 ) / 10;
+		xpos += pointGap;
+	}
+	this._primitive.addTic( false, this._labelColor, [], y );
+});
+
+/**
+ * Add categories for the bars.
+ *
+ * @name addCategory
+ * @memberof Primitive.prototype
+ * @type {Function}
+ * @returns {void}
+ *
+ */
+setNonEnumerableReadOnly(BarPlot.prototype, 'addCategory', function addCategory( BarPositions ) {
+	var temp;
+	var cat;
+	var elm;
+	var i;
+
+	cat = [];
+	for ( i = 0; i < BarPositions.length; i++ ) {
+		// row, col, str
+		elm = new Array( 3 );
+
+		// Taking middle
+		temp = floor( ( BarPositions[i][1] + BarPositions[i][0] ) / 2 );
+		elm[ 0 ] = 0;
+		elm[ 1 ] = temp;
+		elm[ 2 ] = ( i + 1 ).toString();
+		cat.push( elm );
+	}
+	this._primitive.addTic( true, this._labelColor, cat );
+});
+
+/**
+ * @name xaxis
+ * @memberof Primitive.prototype
+ * @type {Array}
+ * @throws {TypeError}
+ *
+ */
+defineProperty( BarPlot.prototype, 'xaxis', {
+	'configurable': false,
+	'enumerable': true,
+	'set': setXaxis,
+	'get': getXaxis
+});
+
+/**
+ * @name yaxis
+ * @memberof Primitive.prototype
+ * @type {Array}
+ * @throws {TypeError}
+ *
+ */
+defineProperty( BarPlot.prototype, 'yaxis', {
+	'configurable': false,
+	'enumerable': true,
+	'set': setYaxis,
+	'get': getYaxis
+});
+
+/**
+ * @name rows
+ * @memberof Primitive.prototype
+ * @type {number}
+ * @throws {TypeError}
+ *
+ */
+defineProperty( BarPlot.prototype, 'rows', {
+	'configurable': false,
+	'enumerable': true,
+	'set': setRows,
+	'get': getRows
+});
+
+/**
+ * @name cols
+ * @memberof Primitive.prototype
+ * @type {number}
+ * @throws {TypeError}
+ *
+ */
+defineProperty( BarPlot.prototype, 'cols', {
+	'configurable': false,
+	'enumerable': true,
+	'set': setCols,
+	'get': getCols
+});
+
+/**
+ * @name barColor
+ * @memberof Primitive.prototype
+ * @type {string}
+ * @throws {TypeError}
+ *
+ */
+defineProperty( BarPlot.prototype, 'barColor', {
+	'configurable': false,
+	'enumerable': true,
+	'set': setBarColor,
+	'get': getBarColor
+});
+
+/**
+ * @name labelColor
+ * @memberof Primitive.prototype
+ * @type {string}
+ * @throws {TypeError}
+ *
+ */
+defineProperty( BarPlot.prototype, 'labelColor', {
+	'configurable': false,
+	'enumerable': true,
+	'set': setLabelColor,
+	'get': getLabelColor
+});
+
+/**
+ * @name marker
+ * @memberof Primitive.prototype
+ * @type {string}
+ * @throws {TypeError}
+ *
+ */
+defineProperty( BarPlot.prototype, 'marker', {
+	'configurable': false,
+	'enumerable': true,
+	'set': setMarker,
+	'get': getMarker
+});
+
+/**
+* Find the best suitable width of the bars according to the width.
+*
+* @private
+*
+* @type {Function}
+* @param {Array} x - date of x axis
+* @param {Array} y - data of y axis
+* @param {number} containerWidth - width of the container
+* @returns {Array} array of scale represent width and height of bar
+* */
+function scaleBar( x, y, containerWidth ) {
```

Comment:
Helper functions such as this should be moved to the top of the file in a `// FUNCTIONS //` section before `// MAIN //`.